### PR TITLE
Feature/shell

### PR DESCRIPTION
**Pull request by gjbex**
A bit of a misnomer, contains latest
  * Bash (GCC)
  * Bison (GCC)
  * flex (GCC)
  * Grep (GCC)
  * Parallel (GCC)
  * OpenSSL (GCC)
  * zlib (GCC)
  * cURL (GCC)
  * cURL (intel)

Note that grep and parallel exist, but but their naming is not consistent, so these versions follow the convention.

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/g/Grep/Grep-2.21-GCC-4.9.2.eb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name = 'Grep'
+version = '2.21'
+easyblock = 'ConfigureMake'
+
+homepage = 'http://http://www.gnu.org/software/grep/'
```
Comment (boegel) [inline easybuild/easyconfigs/g/Grep/Grep-2.21-GCC-4.9.2.eb]: double `http://`, please fix

File: easybuild/easyconfigs/g/Grep/Grep-2.21-GCC-4.9.2.eb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name = 'Grep'
+version = '2.21'
+easyblock = 'ConfigureMake'
```
Comment (boegel) [inline easybuild/easyconfigs/g/Grep/Grep-2.21-GCC-4.9.2.eb]: this line should be first, followed by an empty line

File: easybuild/easyconfigs/c/cURL/cURL-7.40.0-GCC-4.9.2.eb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ SSL certificates, HTTP POST, HTTP PUT, FTP uploading, HTTP form based upload,
+ proxies, cookies, user+password authentication (Basic, Digest, NTLM, Negotiate,
+ Kerberos), file transfer resume, http proxy tunneling and more."""
+
+toolchain = {'name': 'GCC', 'version': '4.9.2'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://curl.haxx.se/download/']
+
+dependencies = [('OpenSSL', '1.0.1k')]
```
Comment (boegel) [inline easybuild/easyconfigs/c/cURL/cURL-7.40.0-GCC-4.9.2.eb]: an `openssl` provided by the OS should be preferred here, cfr. the Python easyconfigs?

File: easybuild/easyconfigs/c/cURL/cURL-7.40.0-intel-2015a.eb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ SSL certificates, HTTP POST, HTTP PUT, FTP uploading, HTTP form based upload,
+ proxies, cookies, user+password authentication (Basic, Digest, NTLM, Negotiate,
+ Kerberos), file transfer resume, http proxy tunneling and more."""
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://curl.haxx.se/download/']
+
+dependencies = [('OpenSSL', '1.0.1k')]
```
Comment (boegel) [inline easybuild/easyconfigs/c/cURL/cURL-7.40.0-intel-2015a.eb]: osdep?

File: easybuild/easyconfigs/p/Parallel/Parallel-20141122-GCC-4.9.2.eb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+name = 'parallel'
```
Comment (boegel) [inline easybuild/easyconfigs/p/Parallel/Parallel-20141122-GCC-4.9.2.eb]: should be `Parallel`

Comment (boegel) [inline easybuild/easyconfigs/p/Parallel/Parallel-20141122-GCC-4.9.2.eb]: `easyblock = 'ConfigureMake'` is missing here?

**Comment by boegel** [comment]
@gjbex: you're right about `Grep` and `Parallel`. Can you also fix the existing easyconfigs by renaming them and fixing the `name` parameter in them?

**Comment by gjbex** [comment]
Test report by @gjbex
**FAILED**
Build succeeded for 8 out of 9 (9 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/ce6c7b2974a1da7ed43b for a full test report.

File: easybuild/easyconfigs/g/Grep/Grep-2.21-GCC-4.9.2.eb (hunk 1 above)
Comment (gjbex) [inline easybuild/easyconfigs/g/Grep/Grep-2.21-GCC-4.9.2.eb]: Done.

File: easybuild/easyconfigs/g/Grep/Grep-2.21-GCC-4.9.2.eb (hunk 2 above)
Comment (gjbex) [inline easybuild/easyconfigs/g/Grep/Grep-2.21-GCC-4.9.2.eb]: Done.

File: easybuild/easyconfigs/p/Parallel/Parallel-20141122-GCC-4.9.2.eb (hunk 5 above)
Comment (gjbex) [inline easybuild/easyconfigs/p/Parallel/Parallel-20141122-GCC-4.9.2.eb]: Done, and done.

**Comment by gjbex** [comment]
@boegel : Hm, of course I could change parallel -> Parallel and grep -> Grep, but wouldn't that suddenly break things for people?  If my users use 'grep', and for some reason I've to reinstall, 'grep' will vanish and be replaced by 'Grep'.
So should I?

File: easybuild/easyconfigs/c/cURL/cURL-7.40.0-GCC-4.9.2.eb (hunk 3 above)
Comment (gjbex) [inline easybuild/easyconfigs/c/cURL/cURL-7.40.0-GCC-4.9.2.eb]: Sorry, UTS, ?que?

**Comment by gjbex** [comment]
Test report by @gjbex
**FAILED**
Build succeeded for 7 out of 9 (9 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/26b23763705ffc8bdcae for a full test report.

**Comment by boegel** [comment]
> @boegel : Hm, of course I could change parallel -> Parallel and grep -> Grep, but wouldn't that 
> suddenly break things for people? If my users use 'grep', and for some reason I've to reinstall, 'grep' will 
> vanish and be replaced by 'Grep'.
> So should I?

I'd say yes, since I consider this a bug fix. That's the only case where we consider breaking backward compatibility to be acceptable. 

Once your users start using Lmod, which features a case-insensitive `module avail`, they won't care too much. ;-)

**Comment by boegel** [comment]
@gjbex: Jenkins isn't happy about your flex easyconfig yet:

```
AsertionError: Filename '../flex-2.5.39-GCC-4.9.2.eb'
 of parsed easconfig matches expected filename 'flex-2.5.38-GCC-4.9.2.eb'
```

(wrong version in filename)

File: easybuild/easyconfigs/c/cURL/cURL-7.40.0-GCC-4.9.2.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/c/cURL/cURL-7.40.0-GCC-4.9.2.eb]: zie https://github.com/hpcugent/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/p/Python/Python-2.7.9-intel-2015a.eb#L22

**Comment by gjbex** [comment]
Test report by @gjbex
**FAILED**
Build succeeded for 7 out of 9 (9 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/4b3e0c0d8b0d7714df0c for a full test report.

**Comment by gjbex** [comment]
Test report by @gjbex
**FAILED**
Build succeeded for 8 out of 9 (9 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/f302f993d6130b6f67c4 for a full test report.

**Comment by gjbex** [comment]
Fixed flex problems as well.

**Comment by gjbex** [comment]
Test report by @gjbex
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/40e560aab4b4f64cf81c for a full test report.

File: easybuild/easyconfigs/c/cURL/cURL-7.40.0-intel-2015a.eb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ Kerberos), file transfer resume, http proxy tunneling and more."""
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://curl.haxx.se/download/']
+
+osdependencies = [('openssl-devel', 'libssl-dev')]
+
+configopts = "--with-ssl=$EBROOTOPENSSL"
```
Comment (boegel) [inline easybuild/easyconfigs/c/cURL/cURL-7.40.0-intel-2015a.eb]: this is pretty useless now the OpenSSL dep is gone

I prefer leaving in the commented out OpenSSL dep though, so people can switch easily.

Comment (gjbex) [inline easybuild/easyconfigs/c/cURL/cURL-7.40.0-intel-2015a.eb]: Ok, reinserted dependency on OpenSSL, but commented out, and commented out configopts for consistency.

**Comment by gjbex** [comment]
Test report by @gjbex
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/24dce65310c1b7231d6d for a full test report.

File: easybuild/easyconfigs/b/Bash/Bash-4.3-GCC-4.9.2.eb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+#
+# Copyright:: Copyright 2012-2013 University of Luxembourg/Computer Science and Communications Research Unit
+# Authors::   Valentin Plugaru <valentin.plugaru@gmail.com>
+# License::   MIT/GPL
+# $Id$
+#
+# This work implements a part of the HPCBIOS project and is a component of the policy:
+# http://hpcbios.readthedocs.org/en/latest/HPCBIOS_05-06.html
+##
+
```
Comment (boegel) [inline easybuild/easyconfigs/b/Bash/Bash-4.3-GCC-4.9.2.eb]: this is missing `easyblock = 'ConfigureMake'` too

the build works with EasyBuild v1.16.1 (cfr. test report), but it won't with EasyBuild v2.0 (and Jenkins will tell you that with failing tests)

Comment (gjbex) [inline easybuild/easyconfigs/b/Bash/Bash-4.3-GCC-4.9.2.eb]: Ok, fixed.

On Sat, Jan 17, 2015 at 12:37 AM, Kenneth Hoste <notifications@github.com>
wrote:

> In easybuild/easyconfigs/b/Bash/Bash-4.3-GCC-4.9.2.eb
> <https://github.com/hpcugent/easybuild-easyconfigs/pull/1308#discussion-diff-23117450>
> :
>
> > @@ -0,0 +1,31 @@
> > +##
> > +# This file is an EasyBuild reciPY as per https://github.com/hpcugent/easybuild
> > +#
> > +# Copyright:: Copyright 2012-2013 University of Luxembourg/Computer Science and Communications Research Unit
> > +# Authors::   Valentin Plugaru <valentin.plugaru@gmail.com>
> > +# License::   MIT/GPL
> > +# $Id$
> > +#
> > +# This work implements a part of the HPCBIOS project and is a component of the policy:
> > +# http://hpcbios.readthedocs.org/en/latest/HPCBIOS_05-06.html
> > +##
> > +
>
> this is missing easyblock = 'ConfigureMake' too
>
> the build works with EasyBuild v1.16.1 (cfr. test report), but it won't
> with EasyBuild v2.0 (and Jenkins will tell you that with failing tests)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcugent/easybuild-easyconfigs/pull/1308/files#r23117450>
> .
>



-- 
- dr. Geert Jan BEX
- HPC analyst/consultant Hasselt University & University of Leuven
- email:                geertjan.bex@uhasselt.be
- homepage:         http://alpha.uhasselt.be/~gjb/
- PGP public key: http://alpha.uhasselt.be/~gjb/pgp.txt
---------------------------------------------------------

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f766d113043a7d8d17ea for a full test report.

**Comment by boegel** [comment]
Looking great/clean now, so going in. Thanks for the effort @gjbex!

**Pull request merged**